truffleruby/src/main/java/org/truffleruby/language/threadlocal/ThreadLocalInFrameNode.java

```diff
@@ -0,0 +1,201 @@
+package org.truffleruby.language.threadlocal;
+
+import org.truffleruby.Layouts;
+import org.truffleruby.RubyContext;
+import org.truffleruby.language.RubyBaseNode;
+import org.truffleruby.language.arguments.RubyArguments;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.Frame;
+import com.oracle.truffle.api.frame.FrameDescriptor;
+import com.oracle.truffle.api.frame.FrameSlot;
+import com.oracle.truffle.api.frame.FrameSlotKind;
+import com.oracle.truffle.api.frame.MaterializedFrame;
+import com.oracle.truffle.api.object.DynamicObject;
+
+public abstract class ThreadLocalInFrameNode extends RubyBaseNode {
+
+    protected final String variableName;
+    protected final int limit;
+
+    public ThreadLocalInFrameNode(String variableName, int limit) {
```

Comment:
I think the limit should be just be in a `getLimit()` method returning `getContext().getOptions().FRAME_VARIABLE_ACCESS_LIMIT`, so it doesn't need to be passed and it's immediately clear what option affects it.